preupg/application.py

```diff
@@ -459,48 +443,40 @@ def prepare_scan_system(self):
         return 0
 
     def is_module_set_valid(self):
-        scenario = self.get_scenario()
-        if scenario is None:
-            log_message('Invalid scenario: %s' % self.conf.contents)
+        if self.module_set_dirname is None:
+            log_message('Invalid scenario: %s' % self.module_set_path)
             return False
-        scenario_path = os.path.join(self.conf.source_dir, scenario)
-        if not os.path.isdir(scenario_path):
-            log_message('Invalid scenario: %s' % scenario,
+        if not os.path.isdir(self.module_set_path):
+            log_message('Invalid scenario: %s' % self.module_set_path,
                         level=logging.ERROR)
             return False
         # Validate properties.ini file in module set dir
         try:
-            module_set_path = self.conf.scan
-            # in case --contents option is used
-            if module_set_path is None:
-                # strip all-xccdf.xml from path
-                module_set_path = os.path.dirname(self.conf.contents)
-                if not os.path.isdir(module_set_path):
-                    module_set_path = os.path.normpath(
-                        os.path.join(self.execution_dir, module_set_path))
-            ModuleSetUtils.get_module_set_os_versions(module_set_path)
+            ModuleSetUtils.get_module_set_os_versions(self.module_set_path)
         except EnvironmentError as err:
             log_message(str(err), level=logging.ERROR)
             return False
         return True
 
     def generate_report(self):
```

Comment:
it should be renamed in future as it does something completely different than it says.

Reply:
Will be dealt with under https://github.com/upgrades-migrations/preupgrade-assistant/issues/332.